src/gateway/bridge/shard_runner.rs

```diff
@@ -165,7 +165,7 @@ impl ShardRunner {
 
             if let Some(event) = event {
                 #[cfg(feature = "collector")]
-                self.collectors.retain_mut(|callback| (callback.0)(&event));
+                self.collectors.lock().expect("poison").retain_mut(|callback| (callback.0)(&event));
```

Comment:
How come you chose to use `std::sync::Mutex` instead of `tokio::sync::Mutex` here? Since this function is `async` doing a synchronous block isn't a great idea. But, this isn't really likely to block, because users aren't going to be registering collectors very often. And, it probably doesn't make sense to make `Shard::add_collector` async.

Reply:
I initially used tokio::sync::Mutex, actually, until I realized it would make the `collector` function (and Shard::add_collector too) async. And I thought about it and none of the code that interacts with collectors ever block - so a non-async Mutex makes more sense here